src/main/java/dev/cdevents/CDEventTypes.java

```diff
@@ -14,217 +18,922 @@ private CDEventTypes() {
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("repository", repository.toString())
+                        .build();
+        return cdToSend;
+    }
+
+    /**
+     * Creates a continuous delivery change reviewed event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     * @param eventType
+     * @param id
+     * @param source
+     * @param repository
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * change reviewed event extensions
+     */
+
+    public static CloudEvent createChangeReviewedEvent(
+            final String eventType, final String id,
+            final URI source, final Repository repository,
+            final String data) {
+        CloudEvent cdToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("repository", repository.toString())
+                        .build();
+        return cdToSend;
+    }
+
+    /**
+     * Creates a continuous delivery change merged event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     * @param eventType
+     * @param id
+     * @param source
+     * @param repository
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * change merged event extensions
+     */
+
+    public static CloudEvent createChangeMergedEvent(
+            final String eventType, final String id,
+            final URI source, final Repository repository,
+            final String data) {
+        CloudEvent cdToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("repository", repository.toString())
+                        .build();
+        return cdToSend;
+    }
+
+    /**
+     * Creates a continuous delivery change abandoned event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     * @param eventType
+     * @param id
+     * @param source
+     * @param repository
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * change abandoned event extensions
+     */
+
+    public static CloudEvent createChangeAbandonedEvent(
+            final String eventType, final String id,
+            final URI source, final Repository repository,
+            final String data) {
+        CloudEvent cdToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("repository", repository.toString())

```

Comment:
please use jackson-databind to write this repository as String, say objectMapper.writeValueAsString(repository)